### DISCUSSION TITLE
Docs proposal: destructing page data breaks updates

### DISCUSSION THREAD

**Discussion by korywka** [sveltejs/kit#10038] [General]
When you start with SvelteKit and have weak knowledge of Svelte (e.g. used it a long time ago), you can easily destruct page data like this:

```js
/** @type {import('./$types').PageData} */
export let data;

const (let) { item1, item2 } = data;
```

and lose all rerenders (updates) for data (`item1`, `item2`).

My proposal is to add notice about this here: https://kit.svelte.dev/docs/load

At least maybe someone have the same problem and will find this thread via search.

**Comment by geoffrich** [comment]
We do have this documented in the state management section of the docs: https://kit.svelte.dev/docs/state-management#component-state-is-preserved

Might be worth linking to that section from the `load` docs, though

**Reply by korywka** [reply]
Heh, it would save me a couple of hours for debug.
